Sources/AsyncHTTPClient/ConnectionPool.swift

```diff
@@ -368,11 +370,57 @@ final class ConnectionPool {
             }
         }
 
+        private func makeNonTSBootstrap(on eventLoop: EventLoop) throws -> NIOClientTCPBootstrap {
+            let tlsConfiguration = configuration.tlsConfiguration ?? TLSConfiguration.forClient()
+            let sslContext = try NIOSSLContext(configuration: tlsConfiguration)
+            let tlsProvider = try NIOSSLClientTLSProvider<ClientBootstrap>(context: sslContext, serverHostname: (key.scheme == .unix || key.host.isIPAddress) ? nil : key.host)
+            return NIOClientTCPBootstrap(ClientBootstrap(group: eventLoop), tls: tlsProvider)
+        }
+        
+        /// create a TCP Bootstrap based off what type of `EventLoop` has been passed to the function.
+        private func makeBootstrap(on eventLoop: EventLoop) throws -> NIOClientTCPBootstrap {
+            let bootstrap: NIOClientTCPBootstrap
+            #if canImport(Network)
+            if #available(OSX 10.14, iOS 12.0, tvOS 12.0, watchOS 6.0, *), eventLoop is NIOTSEventLoop {
+                let tlsProvider = NIOTSClientTLSProvider(tlsOptions: .init())
```

Comment:
it seems we'll need some kind of translation from `TLSConfiguration` to `NWProtocolTLS.Options`, @Lukasa what do you think will be the right approach?

Reply:
Right now I think we'll probably need to hand-roll this here. In an ideal world we would define a shim module, but conditional dependency isn't real right now so we can't do that nicely.